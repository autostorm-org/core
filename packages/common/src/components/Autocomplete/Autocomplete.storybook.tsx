import React from "react";
import Autocomplete from "./Autocomplete";
import { Maybe } from "../../types";
import { Typography, TypographyVariant_ENUM } from "../Typography";
import { IAutoCompleteOption } from "./Autocomplete.types";
import { Icon } from "../Icon";
export default {
  title: "Autocomplete",
  component: Autocomplete,
};

export const AutocompleteStory = () => {
  const [selectedOption, setSelected] = React.useState<
    Maybe<IAutoCompleteOption<HTMLElement>>
  >(null);
  return (
    <div style={{ padding: 10 }}>
      <Typography variant={TypographyVariant_ENUM.h3}>Autocomplete</Typography>
      <Autocomplete
        id={"autocomplete-story"}
        options={[
          {
            label: "Bitcoin",
            value: "BTC",
            icon: () => (
              <Icon
                src={"/crypto-logos/btc.svg"}
                width={22}
                height={22}
                alt={"bitcoin logo"}
                aria-hidden={"true"}
              />
            ),
          },
          {
            label: "Ethereum",
            value: "ETH",
            icon: () => (
              <Icon
                src={"/crypto-logos/eth.svg"}
                width={22}
                height={22}
                alt={"Ethereum logo"}
                aria-hidden={"true"}
              />
            ),
          },
          {
            label: "Binance Coin",
            value: "BNB",
            icon: () => (
              <Icon
                src={"/crypto-logos/bnb.svg"}
                width={22}
                height={22}
                alt={"BNB logo"}
                aria-hidden={"true"}
              />
            ),
          },
        ]}
        selectedOption={selectedOption}
        onSelection={setSelected}
        aria-label={"Cryptocurrencies combobox"}
      />

      <div aria-hidden={true}>
        <Typography variant={TypographyVariant_ENUM.h5}>
          Selected option Label: {selectedOption?.label}
        </Typography>
        <br />
        <Typography variant={TypographyVariant_ENUM.h5}>
          Selected option value: {selectedOption?.value}
        </Typography>
      </div>
    </div>
  );
};
