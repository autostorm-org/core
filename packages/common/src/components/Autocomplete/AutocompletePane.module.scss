@import "./autocomplete-variables.scss";

.root {
  margin: 0px;
  padding: 0px;
}

.listElement {
  list-style-type: none;
  padding: 10px 24px 10px 24px;
  display: flex;
  align-items: center;
  font-size: 14px;
  font-family: var(--autocomplete-pane-font-family);
  font-weight: var(--autocomplete-pane-font-weight);
  background-color: var(--autocomplete-pane-background-color);
  color: var(--autocomplete-pane-color);
  cursor: pointer;
  &:hover {
    background-color: var(--autocomplete-pane-background-color-highlight);
  }
}
.listElementFocused {
  background-color: var(--autocomplete-pane-background-color-highlight);
}
.listElementWrapper {
  margin: 0px 5px 0px 5px;
}
