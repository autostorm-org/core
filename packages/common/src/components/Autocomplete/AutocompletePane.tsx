import React from "react";
import type { IAutoCompleteOption } from "./Autocomplete.types";
import type { IIdentifiable, IAriaLabel } from "../../types";
import styles from "./AutocompletePane.module.scss";
import { ProgressPlugin } from "webpack";

interface AutocompletePaneProps extends IIdentifiable, IAriaLabel {
  options: Array<IAutoCompleteOption<HTMLLIElement>>;
  onSelection?: (e: IAutoCompleteOption<HTMLLIElement>) => void;
  textBuffer?: string;
  focusedIndex?: number;
}

const AutocompletePane = (props: AutocompletePaneProps) => {
  const refs = props.options.map((value, i) => {
    return React.createRef<HTMLLIElement>();
  });

  React.useLayoutEffect(() => {
    if (props.focusedIndex != null && props.focusedIndex > -1) {
      if (refs[props.focusedIndex] != null) {
        refs[props.focusedIndex].current?.scrollIntoView({
          behavior: "smooth",
          block: "start",
        });
      }
    }

    return () => {
      if (refs.length > -1 && refs[0] && refs[0].current) {
        refs[0].current?.scrollIntoView({
          behavior: "smooth",
          block: "start",
        });
      }
    };
  }, [props]);
  return (
    <ul
      className={styles.root}
      id={props.id}
      role={"listbox"}
      tabIndex={-1}
      aria-label={`${props.options.length} combobox options available`}
    >
      {props.options.map((option, i) => {
        return (
          <li
            id={`${option.label} - ${option.value}`}
            role={"option"}
            ref={refs[i]}
            aria-selected={props.focusedIndex == i ? "true" : "false"}
            key={`key ${i} ${option.label}`}
            className={`${styles.listElement} ${
              props.focusedIndex == i ? styles.listElementFocused : ""
            }`}
            onClick={(e) => {
              e.preventDefault();
              if (option.onClick != null) {
                option.onClick(e);
              }

              if (props.onSelection != null) {
                props.onSelection(option);
              }
            }}
          >
            {option.icon && <option.icon />}
            <span className={styles.listElementWrapper}>{option.label}</span>
          </li>
        );
      })}
    </ul>
  );
};
export default AutocompletePane;
export type { AutocompletePaneProps };
