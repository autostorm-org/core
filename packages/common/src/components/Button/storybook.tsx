import React from "react";
import Button from "./Button";

// This default export determines where you story goes in the story list
export default {
  title: "Button",
  component: Button,
};

export const ButtonStory = () => {
  return <Button>Hello</Button>;
};
