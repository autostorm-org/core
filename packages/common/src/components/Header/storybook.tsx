import React from "react";
import Header from "./Header";
import HeaderLogo from "./HeaderLogo";
import { HeaderLocations } from "./constants";
import HeaderItem from "./HeaderItem";
import Spain from "../../assets/flags/Spain.png";
import Brazil from "../../assets/flags/Brazil.png";
import ResponsiveTemplate from "../ResponsiveTemplate";
import ListFixed from "../ListFixed";
import Icon from "../Icon";
import HeaderItemAbsolute from "./HeaderItemAbsolute";
import { ListFullScreen, ListFullScreenRenderer } from "../List";
import { Menu } from "react-feather";
// This default export determines where you story goes in the story list
export default {
  title: "Header",
  component: Header,
};
declare var require: any;

const logoPath = "./../../assets/autofica-defi-day-logo.svg";

const Desktop = (args) => (
  <Header>
    <HeaderLogo
      logoPath={require("./../../assets/autofica-defi-logo-short-6.svg")}
      itemLocation={HeaderLocations.left}
    />
    <HeaderItem>Loan Manager</HeaderItem>
    <HeaderItem>Whitepaper</HeaderItem>
    <HeaderItem
      dropDown
      sections={[
        {
          links: [
            {
              content: (
                <>
                  <span>br-PT</span>{" "}
                  <Icon
                    src={Brazil}
                    width={23}
                    height={13}
                    alt={"portuguese"}
                  />
                </>
              ),
            },
          ],
        },
      ]}
      Submenu={ListFixed}
    >
      es-ES <Icon src={Spain} width={23} height={13} alt={"espanol"} />
    </HeaderItem>
  </Header>
);

const Mobile = (args) => {
  const [isOpen, setOpen] = React.useState(false);
  return (
    <Header
      submenu={
        isOpen ? (
          <ListFullScreen>
            <ListFullScreenRenderer>Vende</ListFullScreenRenderer>{" "}
            <ListFullScreenRenderer>Contacto</ListFullScreenRenderer>
            <ListFullScreenRenderer>Ayuda</ListFullScreenRenderer>
            <ListFullScreenRenderer>Mi Cuenta</ListFullScreenRenderer>
          </ListFullScreen>
        ) : null
      }
    >
      <HeaderLogo
        logoPath={require("./../../assets/autofica-defi-day-logo.svg")}
        itemLocation={HeaderLocations.center}
        overrides={{ paddingTop: 10, paddingBottom: 10, width: 125 }}
      />
      <HeaderItemAbsolute
        itemLocation={HeaderLocations.right}
        onClick={function () {
          setOpen(!isOpen);
        }}
      >
        <Menu color="white" />
      </HeaderItemAbsolute>
    </Header>
  );
};

const ResponsiveStory = () => (
  <ResponsiveTemplate mobile={Mobile} tablet={Desktop} desktop={Desktop} />
);

export const FirstStory = ResponsiveStory.bind({});

FirstStory.args = {
  /* the args you need here will depend on your component */
};
