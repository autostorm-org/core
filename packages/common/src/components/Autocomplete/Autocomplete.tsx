import React from "react";
import { ChevronDown } from "react-feather";
import AutocompletePane from "./AutocompletePane";
import { Input } from "../Input";
import { EnumAutoCompleteStates } from "./Autocomplete.types";
import type {
  IAutoCompleteOption,
  AutocompleteContextValue,
} from "./AutocompleteContext";
import type { IFocusable, IChangeable, IAriaLabel, Maybe } from "../../types";
import styles from "./Autocomplete.module.scss";

interface AutocompleteProps
  extends IFocusable<HTMLInputElement>,
    IChangeable<HTMLInputElement>,
    IAriaLabel {
  options: Array<IAutoCompleteOption<HTMLElement>>;
  selectedOption: Maybe<IAutoCompleteOption<HTMLElement>>;
  onSelection?: (option: Maybe<IAutoCompleteOption<HTMLElement>>) => void;
  id: string;
}

const filterOptions = (
  allOptions: Array<IAutoCompleteOption<HTMLElement>>,
  bufferText: string
): Array<IAutoCompleteOption<HTMLElement>> => {
  return allOptions.filter((option) => {
    return option.label.toLocaleLowerCase().includes(bufferText.toLowerCase());
  });
};

const Autocomplete = (props: AutocompleteProps) => {
  const [state, setState] = React.useState<
    AutocompleteContextValue<HTMLElement>
  >({
    focusState: EnumAutoCompleteStates.OFF,
    selectedOption: props.selectedOption,
  });

  const onInputFocus = React.useCallback(
    (e): void => {
      console.log("FOCUSED");
      if (state.focusState == EnumAutoCompleteStates.OFF) {
        setState({
          focusState: EnumAutoCompleteStates.SELECTING,
          selectedOption: state.selectedOption,
        });
      }

      if (state.focusState == EnumAutoCompleteStates.SELECTED) {
        setState({
          focusState: EnumAutoCompleteStates.SELECTING,
          selectedOption: state.selectedOption,
        });
      }

      e.target.select();
    },
    [state]
  );

  const rootNode = React.useRef<HTMLDivElement | null>(null);
  const inputNode = React.useRef<HTMLLabelElement | null>(null);
  const [focusedIndex, setFocusedIndex] = React.useState<number>(-1);
  const [textBuffer, setTextBuffer] = React.useState<string>("");
  const onSelection = (option: IAutoCompleteOption<HTMLElement>) => {
    setState({
      focusState: EnumAutoCompleteStates.SELECTED,
      selectedOption: option,
    });
    setTextBuffer(option?.label || "");
    setFocusedIndex(-1);
  };
  const filteredOptions = React.useMemo(
    () => filterOptions(props.options, textBuffer),
    [props.options, textBuffer]
  );
  React.useLayoutEffect(() => {
    const inputKeyPressListener = (e: KeyboardEvent) => {
      switch (e.key) {
        case "Down": // IE/Edge specific value
        case "ArrowDown":
          // Do something for "down arrow" key press.
          if (state.focusState == EnumAutoCompleteStates.SELECTING) {
            e.preventDefault();

            if (focusedIndex == filteredOptions.length - 1) {
              setFocusedIndex(-1);
            } else {
              setFocusedIndex(
                Math.min(focusedIndex + 1, filteredOptions.length - 1)
              );
            }
          }

          break;
        case "Up": // IE/Edge specific value
        case "ArrowUp":
          // Do something for "up arrow" key press.
          if (state.focusState == EnumAutoCompleteStates.SELECTING) {
            e.preventDefault();

            if (focusedIndex == -1) {
              setFocusedIndex(filteredOptions.length - 1);
            } else {
              setFocusedIndex(Math.max(focusedIndex - 1, -1));
            }
          }
          break;

        case "Enter":
          // Do something for "enter" or "return" key press.
          if (focusedIndex > -1) {
            onSelection(filteredOptions[focusedIndex]);
          }
          break;
        case "Esc": // IE/Edge specific value
        case "Tab": // IE/Edge specific value
        case "Escape":
          if (state.selectedOption == null) {
            setState({
              focusState: EnumAutoCompleteStates.OFF,
              selectedOption: state.selectedOption,
            });
            setFocusedIndex(-1);
          }

          if (state.selectedOption != null) {
            setState({
              focusState: EnumAutoCompleteStates.SELECTED,
              selectedOption: state.selectedOption,
            });
            setFocusedIndex(-1);
          }

          break;
        default:
          return; // Quit when this doesn't handle the key event.
      }
    };
    inputNode.current?.addEventListener("keydown", inputKeyPressListener);

    return () => {
      inputNode.current?.removeEventListener("keydown", inputKeyPressListener);
    };
  }, [state, focusedIndex, filteredOptions]);
  React.useLayoutEffect(() => {
    const documentClickHandler = (e: MouseEvent): void => {
      const root = rootNode.current;
      // type assertion is inevitable as  EventTarget is the most generic type
      const target = e.target as Node | null;
      if (e.target == null || root == null) {
        return;
      }
      if (root.contains(target) == false) {
        if (state.focusState == EnumAutoCompleteStates.SELECTING) {
          setState({
            focusState:
              state.selectedOption == null
                ? EnumAutoCompleteStates.OFF
                : EnumAutoCompleteStates.SELECTED,
            selectedOption: state.selectedOption,
          });
        }

        if (state.focusState == EnumAutoCompleteStates.SELECTED) {
          setState({
            focusState: EnumAutoCompleteStates.SELECTED,
            selectedOption: state.selectedOption,
          });
        }
      }
      setFocusedIndex(-1);
    };

    document.addEventListener("click", documentClickHandler, false);
    return () => {
      document.removeEventListener("click", documentClickHandler);
    };
  }, [rootNode.current, state.selectedOption, state.focusState, focusedIndex]);
  React.useEffect(() => {
    if (props.onSelection) {
      props.onSelection(state.selectedOption);
    }
  }, [state.selectedOption]);
  return (
    <div className={styles.root} ref={rootNode}>
      <Input
        type="text"
        id={props.id}
        iconLeft={state.selectedOption?.icon}
        iconRight={ChevronDown}
        override={styles.inputOverride}
        onFocusCapture={onInputFocus}
        onChange={(e) => {
          if (e.target.value != state.selectedOption?.label) {
            setState({
              selectedOption: null,
              focusState: EnumAutoCompleteStates.SELECTING,
            });
          }
          setTextBuffer(e.target.value);
          setFocusedIndex(-1);
        }}
        autocomplete={"off"}
        ref={inputNode}
        value={textBuffer}
        role={"combobox"}
        aria-autocomplete="list"
        aria-haspopup="true"
        aria-expanded={
          state.focusState == EnumAutoCompleteStates.SELECTING
            ? "true"
            : "false"
        }
        aria-owns={`${props.id}-pane`}
        aria-activedescendant={
          filteredOptions[focusedIndex]
            ? `${filteredOptions[focusedIndex].label} - ${filteredOptions[focusedIndex].value}`
            : undefined
        }
        aria-label={props["aria-label"]}
      />
      <div
        className={`${styles.paneContainer} ${styles.paneWrapper} ${
          state.focusState == EnumAutoCompleteStates.SELECTING
            ? styles.paneExpanded
            : styles.paneCollapsed
        }`}
      >
        <AutocompletePane
          options={state.selectedOption == null ? filteredOptions : []}
          textBuffer={textBuffer}
          onSelection={onSelection}
          focusedIndex={focusedIndex}
          id={`${props.id}-pane`}
          aria-label={props["aria-label"]}
        />
      </div>
    </div>
  );
};
export default Autocomplete;
export type { AutocompleteProps };
