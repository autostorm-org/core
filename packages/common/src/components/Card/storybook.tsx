import React from "react";
import Card from "./Card";
import Typography, { EnumTypographyVariant } from "../Typography";
import Button from "../Button";
import { withKnobs, boolean } from "@storybook/addon-knobs";
import { ThemeContext, nightTheme, defaultTheme } from "../../theme";
import ResponsiveTemplate from "../ResponsiveTemplate";
// This default export determines where you story goes in the story list
export default {
  title: "Card",
  component: Card,
  decorators: [withKnobs],
};

const MobileCard = () => {
  return (
    <div style={{ height: 184, width: 350 }}>
      <Card>
        <Typography variant={EnumTypographyVariant.h3}>Card title</Typography>
        <br />
        <Typography variant={EnumTypographyVariant.p}>
          Card body contents. Card body contents. Card body contents. Card body
          contents.
        </Typography>
        <br />
        <Button onClick={() => {}} overrides={{ marginTop: 9 }}>
          Call to action
        </Button>
      </Card>
    </div>
  );
};

const DesktopCard = () => {
  return (
    <div style={{ height: 240, width: 560 }}>
      <Card style={{ padding: "30px" }}>
        <Typography variant={EnumTypographyVariant.h1}>
          Crypto car loan finder
        </Typography>
        <br />
        <Typography variant={EnumTypographyVariant.p}>
          Get a collateralized loan for your next car purchase. Chose your
          backing assets. Powered by Multiplier.{" "}
        </Typography>
        <br />
        <Button onClick={() => {}} overrides={{ marginTop: 10 }}>
          Call to action
        </Button>
      </Card>
    </div>
  );
};

const ResponsiveStory = () => {
  const isDayMode = boolean("DAY MODE", true);
  return (
    <ThemeContext.Provider value={isDayMode ? defaultTheme : nightTheme}>
      <ResponsiveTemplate
        mobile={MobileCard}
        desktop={DesktopCard}
        tablet={DesktopCard}
      />
    </ThemeContext.Provider>
  );
};

export const FirstStory = ResponsiveStory.bind({});
