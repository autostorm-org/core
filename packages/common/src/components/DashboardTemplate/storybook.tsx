import React from "react";
import DashboardTemplate from "./DashboardTemplate";
import { BoundedRow } from "../BoundedRow";
import { Typography, TypographyVariant_ENUM } from "../Typography";
import Icon from "../Icon";
import Spain from "../../assets/flags/Spain.png";
import { HeaderResponsive } from "../Header";
import DashboardOptionsColumn from "./DashboardOptionsColumn";
import DashboardOptionsItem from "./DashboardOptionsItem";
import { PillButton } from "./../PillButton";
import { Eye, ArrowLeftCircle, DollarSign, ArrowUp, Key } from "react-feather";

export default {
  title: "Dashboard Template Stories",
};

export const DashboardOptionsColumnStory = () => {
  return (
    <DashboardOptionsColumn>
      <DashboardOptionsItem activeElement href={"#"}>
        hello now
      </DashboardOptionsItem>
      <DashboardOptionsItem href={"#"}>hello now</DashboardOptionsItem>
      <DashboardOptionsItem href={"#"}>hello now</DashboardOptionsItem>
    </DashboardOptionsColumn>
  );
};

export const DashboardTemplateStory = () => {
  return (
    <>
      <HeaderResponsive
        options={[
          { content: "Loan Manager", href: "#" },
          { content: "Whitepaper", href: "#" },
          {
            content: (
              <>
                es-ES{" "}
                <Icon src={Spain} width={23} height={13} alt={"espanol"} />
              </>
            ),
          },
        ]}
      />
      <div style={{ marginTop: 30 }}>
        <BoundedRow>
          <DashboardTemplate
            mainContent={
              <Typography
                variant={TypographyVariant_ENUM.h3}
                style={{ padding: 30 }}
              >
                Main Content
              </Typography>
            }
            optionsContent={
              <>
                <DashboardOptionsColumn style={{ borderTopLeftRadius: 8 }}>
                  <div style={{ marginTop: 30, marginBottom: 20 }}>
                    <PillButton
                      onClick={() => {
                        console.log("CLIICK");
                      }}
                    >
                      <Icon
                        src={"crypto-logos/bsc.svg"}
                        height={22}
                        width={22}
                        alt={"BSC logo"}
                        style={{ marginRight: 10 }}
                      />
                      BSC Mainnet
                    </PillButton>
                  </div>

                  <DashboardOptionsItem href={"#"} activeElement>
                    <Eye style={{ marginRight: 10 }} />
                    Loans Overview
                  </DashboardOptionsItem>
                  <DashboardOptionsItem href={"#"}>
                    <ArrowLeftCircle style={{ marginRight: 10 }} />
                    Borrow
                  </DashboardOptionsItem>
                  <DashboardOptionsItem href={"#"}>
                    <DollarSign style={{ marginRight: 10 }} />
                    Payback
                  </DashboardOptionsItem>
                  <DashboardOptionsItem href={"#"}>
                    <ArrowUp style={{ marginRight: 10 }} />
                    Withdraw Collateral
                  </DashboardOptionsItem>
                  <DashboardOptionsItem href={"#"}>
                    <Key style={{ marginRight: 10 }} />
                    Deposit
                  </DashboardOptionsItem>
                </DashboardOptionsColumn>
              </>
            }
          />
        </BoundedRow>
      </div>
    </>
  );
};
