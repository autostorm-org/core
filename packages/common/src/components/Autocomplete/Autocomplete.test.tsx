import React from "react";
import renderer from "react-test-renderer";
import { AutocompleteStory } from "./Autocomplete.storybook";
describe("AutocompleteStory Test", () => {
  it("Should render AutocompleteStory component", () => {
    const root = renderer.create(<AutocompleteStory />);
    let tree = root.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
