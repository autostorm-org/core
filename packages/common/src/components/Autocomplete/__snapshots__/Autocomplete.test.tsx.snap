// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutocompleteStory Test Should render AutocompleteStory component 1`] = `
<div
  style={
    Object {
      "padding": 10,
    }
  }
>
  <span
    className="h3"
  >
    Autocomplete
  </span>
  <div
    className="root"
  >
    <label
      aria-label="Cryptocurrencies combobox"
      className="root inputOverride"
      htmlFor="autocomplete-story"
    >
      <input
        aria-autocomplete="list"
        aria-expanded="false"
        aria-haspopup="true"
        aria-owns="autocomplete-story-pane"
        autoComplete="off"
        className="input"
        id="autocomplete-story"
        onChange={[Function]}
        onFocusCapture={[Function]}
        role="combobox"
        type="text"
        value=""
      />
      <svg
        fill="none"
        height={24}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        viewBox="0 0 24 24"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="6 9 12 15 18 9"
        />
      </svg>
    </label>
    <div
      className="paneContainer paneWrapper paneCollapsed"
    >
      <ul
        aria-label="3 combobox options available"
        className="root"
        id="autocomplete-story-pane"
        role="listbox"
        tabIndex={-1}
      >
        <li
          aria-selected="false"
          className="listElement "
          id="Bitcoin - BTC"
          onClick={[Function]}
          role="option"
        >
          <img
            alt="bitcoin logo"
            aria-hidden="true"
            className=""
            height={22}
            src="/crypto-logos/btc.svg"
            width={22}
          />
          <span
            className="listElementWrapper"
          >
            Bitcoin
          </span>
        </li>
        <li
          aria-selected="false"
          className="listElement "
          id="Ethereum - ETH"
          onClick={[Function]}
          role="option"
        >
          <img
            alt="Ethereum logo"
            aria-hidden="true"
            className=""
            height={22}
            src="/crypto-logos/eth.svg"
            width={22}
          />
          <span
            className="listElementWrapper"
          >
            Ethereum
          </span>
        </li>
        <li
          aria-selected="false"
          className="listElement "
          id="Binance Coin - BNB"
          onClick={[Function]}
          role="option"
        >
          <img
            alt="BNB logo"
            aria-hidden="true"
            className=""
            height={22}
            src="/crypto-logos/bnb.svg"
            width={22}
          />
          <span
            className="listElementWrapper"
          >
            Binance Coin
          </span>
        </li>
      </ul>
    </div>
  </div>
  <div
    aria-hidden={true}
  >
    <span
      className="h5"
    >
      Selected option Label: 
    </span>
    <br />
    <span
      className="h5"
    >
      Selected option value: 
    </span>
  </div>
</div>
`;
